osu.Game/Graphics/ParticleSpewer.cs

```diff
@@ -0,0 +1,208 @@
+                    if (timeSinceStart < 0) continue;
+
+                    var alpha = p.AlphaAtTime(timeSinceStart);
+                    if (alpha <= 0) continue;
+
+                    var pos = p.PositionAtTime(timeSinceStart, gravity, maxDuration);
+                    var scale = p.ScaleAtTime(timeSinceStart);
+                    var angle = p.AngleAtTime(timeSinceStart);
+
+                    var rect = createDrawRect(pos, scale);
+
+                    var quad = new Quad(
+                        transformPosition(rect.TopLeft, rect.Centre, angle),
+                        transformPosition(rect.TopRight, rect.Centre, angle),
+                        transformPosition(rect.BottomLeft, rect.Centre, angle),
+                        transformPosition(rect.BottomRight, rect.Centre, angle)
+                    );
+
+                    DrawQuad(Texture, quad, DrawColourInfo.Colour.MultiplyAlpha(alpha), null, vertexAction,
+                        new Vector2(InflationAmount.X / DrawRectangle.Width, InflationAmount.Y / DrawRectangle.Height),
+                        null, TextureCoords);
+                }
+            }
+
+            private RectangleF createDrawRect(Vector2 position, float scale)
+            {
+                var width = Texture.DisplayWidth * scale;
+                var height = Texture.DisplayHeight * scale;
+
+                if (relativePositionAxes.HasFlagFast(Axes.X))
+                    position.X *= sourceSize.X;
+                if (relativePositionAxes.HasFlagFast(Axes.Y))
+                    position.Y *= sourceSize.Y;
+
+                return new RectangleF(
+                    position.X - width / 2,
+                    position.Y - height / 2,
+                    width,
+                    height);
+            }
+
+            private Vector2 transformPosition(Vector2 pos, Vector2 centre, float angle)
+            {
+                float cos = MathF.Cos(angle);
+                float sin = MathF.Sin(angle);
+
+                float x = centre.X + (pos.X - centre.X) * cos + (pos.Y - centre.Y) * sin;
+                float y = centre.Y + (pos.Y - centre.Y) * cos - (pos.X - centre.X) * sin;
+
+                return Vector2Extensions.Transform(new Vector2(x, y), DrawInfo.Matrix);
+            }
+        }
+
+        #endregion
+
+        public struct FallingParticle
+        {
+            public float StartTime;
+            public Vector2 StartPosition;
+            public Vector2 Velocity;
+            public float Duration;
+            public float StartAngle;
+            public float EndAngle;
+            public float EndScale;
+
+            public float AlphaAtTime(float timeSinceStart) => 1 - progressAtTime(timeSinceStart);
+
+            public float ScaleAtTime(float timeSinceStart) => 1 + (EndScale - 1) * progressAtTime(timeSinceStart);
+
+            public float AngleAtTime(float timeSinceStart) => StartAngle + (EndAngle - StartAngle) * progressAtTime(timeSinceStart);
```

Comment:
Opportunity to use `Interpolation.Lerp` here (unless inlining like that is faster? but not sure it would be)

Reply:
`Interpolation.Lerp` only works with doubles so there has to be a cast to float, but that's most likely a negligible performance difference so I went with it.